bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/welcome/NAWelcomePersonHandler.java

```diff
@@ -70,32 +65,34 @@ public void updateChannels(Object module) {
                         lastEvent = event;
                     }
                 });
-            }
+            });
 
             setRefreshRequired(false);
         }
         super.updateChannels(module);
     }
 
     @Override
-    protected State getNAThingProperty(@NonNull String channelId) {
-        NAWelcomePerson module = this.module;
+    protected State getNAThingProperty(String channelId) {
         switch (channelId) {
             case CHANNEL_WELCOME_PERSON_LASTSEEN:
-                return module != null ? toDateTimeType(module.getLastSeen(), timeZoneProvider.getTimeZone())
-                        : UnDefType.UNDEF;
+                return getModule().map(m -> toDateTimeType(m.getLastSeen(), timeZoneProvider.getTimeZone()))
+                        .orElse(UnDefType.UNDEF);
             case CHANNEL_WELCOME_PERSON_ATHOME:
-                return module != null ? module.getOutOfSight() ? OnOffType.OFF : OnOffType.ON : UnDefType.UNDEF;
+                return getModule()
+                        .map(m -> m.getOutOfSight() != null ? toOnOffType(!m.getOutOfSight()) : UnDefType.UNDEF)
+                        .orElse(UnDefType.UNDEF);
             case CHANNEL_WELCOME_PERSON_AVATAR_URL:
                 return toStringType(getAvatarURL());
             case CHANNEL_WELCOME_PERSON_AVATAR:
                 return getAvatarURL() != null ? HttpUtil.downloadImage(getAvatarURL()) : UnDefType.UNDEF;
             case CHANNEL_WELCOME_PERSON_LASTMESSAGE:
-                return (lastEvent != null && lastEvent.getMessage() != null)
-                        ? toStringType(lastEvent.getMessage().replace("<b>", "").replace("</b>", ""))
+                return (getLastEvent().isPresent() && getLastEvent().get().getMessage() != null)
+                        ? toStringType(getLastEvent().get().getMessage().replace("<b>", "").replace("</b>", ""))
```

Comment:
you get the idea, fix wherever else it might apply.

Reply:
Understood